pkg/scheduler/scheduler.go

```diff
@@ -220,6 +221,26 @@ func (s *Scheduler) schedule(ctx context.Context) wait.SpeedSignal {
 		ctx := ctrl.LoggerInto(ctx, log)
 
 		mode := e.assignment.RepresentativeMode()
+
+		if workload.HasTopologyAssignmentWithNodeToReplace(e.Obj) && mode != flavorassigner.Fit && features.Enabled(features.TASFailedNodeReplacementFailFast) {
+			// evict workload we couldn't find the replacement for
+			log.V(2).Info("Evicting workload after failed try to find a node replacement; TASFailedNodeReplacementFailFast enabled")
+			msg := fmt.Sprintf("Workload was evicted as there was no replacement for a failed node: %s", workload.NodeToReplace(e.Obj))
+			if err := workload.EvictWorkload(ctx, s.client, s.recorder, e.Obj, kueue.WorkloadEvictedDueToNodeFailures, msg, s.clock); err != nil {
+				log.V(2).Error(err, "Failed to evict workload after failed try to find a node replacement")
+				continue
+			}
+			if err := workload.RemoveAnnotation(ctx, s.client, e.Obj, kueuealpha.NodeToReplaceAnnotation); err != nil {
+				log.V(2).Error(err, fmt.Sprintf("Failed to remove annotation for node replacement %s", kueuealpha.NodeToReplaceAnnotation))
+				continue
+			}
+			// remove the wl from entries so it's not requeued after we evicted it
```

Comment:
is it really needed? it seems evicted workload would not get requeued for second pass anyway as we check this in `NeedsSecondPass` function?

Reply:
Agree, reverted deleting from the entries